pkg/sentry/platform/kvm/machine.go

```diff
@@ -236,7 +237,11 @@ func newMachine(vm int) (*machine, error) {
 			}
 
 			// Ensure the physical range is mapped.
-			m.mapPhysical(physical, length)
+			if !vr.accessType.Write && arch.Host.String() == "arm64" {
```

Comment:
Hm.. I'm not sure about this. These protection settings on a given region can change. Does this fail in some particular way?

Also, I think that the use of flags here is probably not ideal. There are other options:

In an _arm64 file, define something like:

const (
   readOnlyPhysicalFlags = _KVM_MEM_READONLY
   normalPhysicalFlags = 0
)

And similarly in an amd64 file, define something like:

const (
   readOnlyPhysiscalFlags = 0
   normalPhysicalFlags = 0
)

Then pass in only the read-only status, e.g.

m.mapPhysical(physical, length, !vr.accessType.Write)

.. and plumb the bool through handleBluepillFault as well.

Then have handleBluepillFault do:

var flags = normalPhysicalFlags
if readOnly {
   flags = readOnlyPhysicalFlags
}
errno := m.setMemoryRegion(int(slot), physicalStart, length, virtualStart, flags)

Otherwise passing a random, untyped integer through those layers is very confusing.

Or, instead of a bool, you could even pass the full usermem.AccessType from the virtual region.

Reply:
Hi @amscanne 
I have updated the code logic. Please check it.

In order to unify the platform, "mapPhysical" is divided into 2 parts. One for read-only physicalRegions, and another for flags-none physicalRegions.

On x86 platform, the flags for "setMemoryRegion" can always be set as 0.
There is no need to set read-only physicalRegions. 
So "physicalRegions_rdonly" can be set as nil, and "physicalRegions_available" is equal to the global variable “physicalRegions”.